RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -483,38 +421,71 @@ class PumpOpsSynchronous {
             }
 
             let page = try HistoryPage(pageData: pageData, pumpModel: pumpModel)
+            
+            let newEvents = convertPumpEventToTimestampedEvents(pumpEvents: page.events.reversed(), startDate: startDate, pumpModel: pumpModel)
+            
+            events.append(contentsOf: newEvents.0)
+            
+            if newEvents.cancelled {
+                break
+            }
+        }
+        return (events, pumpModel)
+    }
 
-            for event in page.events.reversed() {
-                if let event = event as? TimestampedPumpEvent, !seenEventData.contains(event.rawData) {
-                    seenEventData.insert(event.rawData)
-
-                    var timestamp = event.timestamp
-                    timestamp.timeZone = pump.timeZone
-
-                    if let date = timestamp.date?.addingTimeInterval(timeAdjustmentInterval) {
-                        if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
-                            NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
-                            break pages
-                        } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
-                            NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
-                            break pages
-                        } else {
-                            if (date.compare(startDate) != .orderedAscending) {
-                                timeCursor = date
-                            }
-                            events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), at: 0)
+    internal func convertPumpEventToTimestampedEvents(pumpEvents: [PumpEvent], startDate: Date, checkDate: Date = Date(), pumpModel: PumpModel) -> (events: [TimestampedHistoryEvent], cancelled: Bool) {
```

Comment:
Could you add documentation to this method?  It does more than just convert pump events to timestamped pump events.  The documentation should indicate that it tries to include all events with a date of startDate or newer, and returns a flag indicating when it is confident it has looked far enough back in history.